### PR TITLE
[GNA] Tests for clamp layer

### PR DESCRIPTION
**Pull request by admitriev-gna**
### Details:
 - Added test for clamp layer to FQ

### CODE REVIEW

File: inference-engine/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/clamp_fq.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        {"sw_fp32", configFP32},
+        {"sw_exact", configSWExact}
+};
+
+const std::vector<std::vector<size_t>> inputShapes = {
+        {1, 250}, {1, 640}, {1, 1024}
+};
+
+const std::vector<size_t> levels_fp = {65535};
+const std::vector<std::vector<float>> inputParams = {{-16, 16, 1}, {-40, 40, 1}};
```
[the remark this answers is not in the document]
Comment (bsochack) [reply inference-engine/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/clamp_fq.cpp:40]: You should also have more ranges of values that are generated. You should also test different clamp ranges than inputs.

**Review by bsochack — CHANGES_REQUESTED** [review]

File: inference-engine/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/clamp_fq.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        {"sw_fp32", configFP32},
+        {"sw_exact", configSWExact}
+};
+
+const std::vector<std::vector<size_t>> inputShapes = {
+        {1, 250}, {1, 640}, {1, 1024}
+};
+
+const std::vector<size_t> level = {65535};
+const std::vector<std::vector<float>> inputParams = {{-16, 16, 1},
```
Comment (bsochack) [inline inference-engine/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/clamp_fq.cpp:40]: limit input params to be within -16 to 16 -> you should also check input ranges like <-10,10>, <-14, 14> etc.

File: inference-engine/src/gna_plugin/gna_plugin_log.hpp [hunk 3]
```diff
@@ -8,7 +8,7 @@
 #include <ie_common.h>
 #include <details/ie_exception.hpp>
 
-// #define GNA_DEBUG
+ #define GNA_DEBUG
```
Comment (nkogteva) [inline inference-engine/src/gna_plugin/gna_plugin_log.hpp:11]: Please remove that change

File: inference-engine/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/clamp_fq.cpp [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+const std::vector<std::vector<size_t>> inputShapes = {
+        {1, 250},
+        {1, 640},
+        {1, 1024}
+};
+
+const std::vector<size_t> level = {65535};
+const std::vector<std::vector<float>> inputParams = {
+                                                     {-16, 16, 1},
```
Comment (bsochack) [inline inference-engine/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/clamp_fq.cpp:43]: I still don't understand. This test should fail if i.e. input range is <-40, 40> and clamp minmax is set i.e. to <-20, 20>. Values outside of <16, 16> will be clamped by hardcoded scale factor = 2048.

Comment (admitriev-gna) [reply inference-engine/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/clamp_fq.cpp:43]: If clamp without FQ than this case will be failed.

**Review by bsochack — APPROVED** [review]

**Pull request merged**